ibis/util.py

```diff
@@ -678,3 +678,28 @@ def __ne__(self, other):
             return self.obj != other.obj
         else:
             return NotImplemented
+
+
+def chunks(n: int, *, chunk_size: int) -> Iterator[tuple[int, int]]:
+    """Return an iterator of chunk start and end indices.
+
+    Parameters
+    ----------
+    n
+        The total number of elements.
+    chunk_size
+        The size of each chunk.
+
+    Returns
+    -------
+    int
+        THE start and end indices of each chunk.
```

Comment:
```suggestion
    tuple[int, int]
        The start and end indices of each chunk.
```

(unless you really wanted that emphasis)

Reply:
😂